vermouth/tests/test_utils.py

```diff
@@ -0,0 +1,136 @@
+
+import string
+import pytest
+from hypothesis import given, strategies, example
+import numpy as np
+from numpy.testing import assert_allclose
+from vermouth import utils
+
+
+@pytest.mark.parametrize('iter_type', (list, tuple, iter, np.array))
+@pytest.mark.parametrize('values, ref', (
+    # All positive integers, one maximum
+    ([1, 2, 3, 4], [4]),
+    ([4, 2, 1, 3], [4]),
+    # All positive integers, multiple maxima
+    ([4, 1, 4, 2], [4] * 2),
+    ([4, 4, 4, 4], [4] * 4),
+    # All positive or negative integers, multiple maxima
+    ([-1, 2, -4, 0, 2], [2] * 2),
+    ([-1, -2, -3, -1], [-1] * 2),
+    # All floats
+    ([0.1, 0.4, 3.4, 2.1, 3.4], [3.4] * 2),
+    # Strings
+    (['z', 'aaa', 'bbb', 'z', 'z'], ['z'] * 3),
+))
+def test_maxes_simple_no_key(values, ref, iter_type):
+    assert utils.maxes(iter_type(values)) == ref
+
+
+@pytest.mark.parametrize('iter_type', (list, tuple, iter, np.array))
+@pytest.mark.parametrize('values, ref', (
+    # All positive integers, one maximum
+    ([1, 2, 3, 4], [1]),
+    ([4, 2, 1, 3], [1]),
+    # All positive integers, multiple maxima
+    ([4, 1, 1, 2], [1] * 2),
+    ([4, 4, 4, 4], [4] * 4),
+    # All positive or negative integers, multiple maxima
+    ([-1, 2, -4, 0, -4], [-4] * 2),
+    ([-3, -2, -3, -1], [-3] * 2),
+    # All floats
+    ([0.1, 0.4, 3.4, 0.1, 3.4], [0.1] * 2),
+))
+def test_maxes_key(values, ref, iter_type):
+       assert utils.maxes(iter_type(values), key=lambda x: -x) == ref
+
+
+@pytest.mark.parametrize('alpha', string.ascii_letters)
+@pytest.mark.parametrize('template', (
+    '{}bc', ' {}bc', '123{}WS', '@#$%{}*&^', '\U0001F62E{}',
+))
+def test_first_alpha(template, alpha):
+    value = template.format(alpha)
+    assert utils.first_alpha(value) == alpha
+
+
+@pytest.mark.parametrize('value', ('', '0123', '\U0001F62E'))
+def test_first_alpha_value_error(value):
+    with pytest.raises(ValueError):
+        utils.first_alpha(value)
+
+
+@pytest.mark.parametrize('iter_type', (list, tuple, iter, np.array))
+@pytest.mark.parametrize('values, ref', (
+    # All equal
+    ([0] * 5, True), ([True] * 4, True),
+    ([None] * 5, True), ([1e-3] * 4, True), 
+    # Not all equal
+    ([0, 1] * 4, False), (['a', 'b', 'c'], False),
+    # Empty
+    ([], True),
+    # One element
+    ([1], True), ([None], True),
+    # Should it be this way?
+    ([0, False] * 3, True),
+))
+def test_are_all_equal(values, ref, iter_type):
+    assert utils.are_all_equal(iter_type(values)) is ref
+
+
+@pytest.mark.parametrize('values, ref', (
+    ([list([0, 1, 2]) for _ in range(3)], True),   
+    ([[0, 1, 2], [3, 4, 5], [6, 7, 8]], False),
+))
+def test_are_all_equal_multidim_list(values, ref):
+    assert utils.are_all_equal(values) is ref
+
+
+def test_are_all_equal_not_implemented():
+    value = np.zeros((3, 4))
+    with pytest.raises(NotImplementedError):
+        utils.are_all_equal(value)
+
+
+@strategies.builds
+def vector_with_random_distance():
```

Comment:
You're *probably* better of using a composite instead of a `builds`. You could ask by making an issue on the hypothesis github